app/views/projects/projects/show.html.erb

```diff
@@ -93,9 +93,11 @@
             <p><%= @project.owner.about_me %></p>
           <% end %>
 
-           <%= link_to contact_user_path(url_name: @project.owner.url_name),
-               class: 'button-secondary' do %>
-            <%= _('Contact {{recipient}}', recipient: @project.owner.name) %>
+          <% if can? :contact_owner, @project %>
+            <%= link_to contact_user_path(url_name: @project.owner.url_name),
+                class: 'button-secondary' do %>
+              <%= _('Contact {{recipient}}', recipient: @project.owner.name) %>
```

Comment:
Haven't investigated why but doesn't seem to actually prevent the contact link showing.

![Screenshot 2020-07-09 at 17 51 11](https://user-images.githubusercontent.com/282788/87067980-d3618080-c20c-11ea-95e9-28dd24b17c4a.png)